include/GafferImage/ImageReader.h

```diff
@@ -110,10 +110,18 @@ class ImageReader : public ImageNode
 		Gaffer::StringPlug *colorSpacePlug();
 		const Gaffer::StringPlug *colorSpacePlug() const;
 
+		Gaffer::StringPlug *colorSpaceMetadataNamePlug();
```

Comment:
`In file included from /home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/intrusive_ptr.hpp:16:0,
                 from /home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/IECore/RefCounted.h:44,
                 from /home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/IECore/RunTimeTyped.h:39,
                 from /home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/IECore/Object.h:44,
                 from include/Gaffer/ValuePlug.h:41,
                 from include/Gaffer/StringPlug.h:41,
                 from src/GafferImage/ImageReader.cpp:41:
/home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/smart_ptr/intrusive_ptr.hpp: In instantiation of 'boost::intrusive_ptr<T>& boost::intrusive_ptr<T>::operator=(const boost::intrusive_ptr<U>&) [with U = const IECore::CompoundObject; T = IECore::Data]':
src/GafferImage/ImageReader.cpp:281:31:   required from here
/home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/smart_ptr/intrusive_ptr.hpp:104:9: error: no matching function for call to 'boost::intrusive_ptr<IECore::Data>::intrusive_ptr(const boost::intrusive_ptr<const IECore::CompoundObject>&)'
         this_type(rhs).swap(*this);
         ^
/home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/smart_ptr/intrusive_ptr.hpp:104:9: note: candidates are:
/home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/smart_ptr/intrusive_ptr.hpp:90:5: note: boost::intrusive_ptr<T>::intrusive_ptr(const boost::intrusive_ptr<T>&) [with T = IECore::Data]
     intrusive_ptr(intrusive_ptr const & rhs): px( rhs.px )
     ^
/home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/smart_ptr/intrusive_ptr.hpp:90:5: note:   no known conversion for argument 1 from 'const boost::intrusive_ptr<const IECore::CompoundObject>' to 'const boost::intrusive_ptr<IECore::Data>&'
In file included from /home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/intrusive_ptr.hpp:16:0,
                 from /home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/IECore/RefCounted.h:44,
                 from /home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/IECore/RunTimeTyped.h:39,
                 from /home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/IECore/Object.h:44,
                 from include/Gaffer/ValuePlug.h:41,
                 from include/Gaffer/StringPlug.h:41,
                 from src/GafferImage/ImageReader.cpp:41:
/home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/smart_ptr/intrusive_ptr.hpp:76:5: note: template<class U> boost::intrusive_ptr<T>::intrusive_ptr(const boost::intrusive_ptr<U>&, typename boost::detail::sp_enable_if_convertible<Y, T>::type)
     intrusive_ptr( intrusive_ptr<U> const & rhs, typename boost::detail::sp_enable_if_convertible<U,T>::type = boost::detail::sp_empty() )
     ^
/home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/smart_ptr/intrusive_ptr.hpp:76:5: note:   template argument deduction/substitution failed:
/home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/smart_ptr/intrusive_ptr.hpp: In substitution of 'template<class U> boost::intrusive_ptr<T>::intrusive_ptr(const boost::intrusive_ptr<U>&, typename boost::detail::sp_enable_if_convertible<Y, T>::type) [with U = const IECore::CompoundObject]':
/home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/smart_ptr/intrusive_ptr.hpp:104:9:   required from 'boost::intrusive_ptr<T>& boost::intrusive_ptr<T>::operator=(const boost::intrusive_ptr<U>&) [with U = const IECore::CompoundObject; T = IECore::Data]'
src/GafferImage/ImageReader.cpp:281:31:   required from here
/home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/smart_ptr/intrusive_ptr.hpp:76:5: error: no type named 'type' in 'struct boost::detail::sp_enable_if_convertible<const IECore::CompoundObject, IECore::Data>'
/home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/smart_ptr/intrusive_ptr.hpp: In instantiation of 'boost::intrusive_ptr<T>& boost::intrusive_ptr<T>::operator=(const boost::intrusive_ptr<U>&) [with U = const IECore::CompoundObject; T = IECore::Data]':
src/GafferImage/ImageReader.cpp:281:31:   required from here
/home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/smart_ptr/intrusive_ptr.hpp:66:5: note: boost::intrusive_ptr<T>::intrusive_ptr(T*, bool) [with T = IECore::Data]
     intrusive_ptr( T * p, bool add_ref = true ): px( p )
     ^
/home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/smart_ptr/intrusive_ptr.hpp:66:5: note:   no known conversion for argument 1 from 'const boost::intrusive_ptr<const IECore::CompoundObject>' to 'IECore::Data*'
/home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/smart_ptr/intrusive_ptr.hpp:62:5: note: boost::intrusive_ptr<T>::intrusive_ptr() [with T = IECore::Data]
     intrusive_ptr() BOOST_NOEXCEPT : px( 0 )
     ^
/home/gafferdev/dev/gaffer/gafferDependencies-0.33.0.0-linux/include/boost/smart_ptr/intrusive_ptr.hpp:62:5: note:   candidate expects 0 arguments, 1 provided
`

Maybe I'm missing something?How do you cast from a specific type ( like `StringData`) to a generic `Data`